dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -223,21 +277,40 @@ DecodeResult LibTiffDecoderInstance::Decode(SampleView<CPUBackend> out, ImageSou
   TensorShape<> out_line_shape = {roi.shape()[1], out_channels};
   TensorShape<> out_line_strides = kernels::GetStrides(out_line_shape);
 
+  // We choose smallest possible type
+  size_t in_type_bits;
+  if (info.bit_depth <= 8) in_type_bits = 8;
+  else if (info.bit_depth <= 16) in_type_bits = 16;
+  else if (info.bit_depth <= 32) in_type_bits = 32;
+  else
+    DALI_FAIL(make_string("Unsupported bit depth: ", info.bit_depth));
+
   TYPE_SWITCH(out.type(), type2id, OutType, (IMGCODEC_TYPES), (
-    VALUE_SWITCH(info.bit_depth, Depth, (8, 16, 32), (
-      using InputType = detail::depth2type<Depth>::type;
-      auto *row_in  = static_cast<InputType *>(row_buf.get());
+    VALUE_SWITCH(in_type_bits, InTypeBits, (8, 16, 32), (
+      using InputType = detail::depth2type<InTypeBits>::type;
+
+      std::vector<InputType> unpacked;
```

Comment:
nitpick: could also use a dynamic scratchpad to get a temporary buffer

Reply:
Cool! I'll do that